torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -351,6 +351,16 @@ ReductionParams OuterNormalizationHeuristic(
   remainder_in_output = ceilDiv(num_outputs_for_reduction, bdimx);
   remainder_in_reduction = ceilDiv(remainder_in_reduction, bdimy);
 
+  // Try to fill the width of the machine by adjust the
+  // the threads devoted to output vs threads used to calculate
+  // the reduction
+  if (remainder_in_output < device_multiprocessor_count) {
```

Comment:
Shouldn't this go above the setting of `remainder_in_output` and `remainder_in_reduction`?

Reply:
I removed the code that was in this location and pushed it up prior to `remainder_in_output` and `remainder_in_reduction` was set as suggested.  I cleaned up the code such that there is a minimum outputs per block that is useful.  Otherwise, pulling from the same cache sector from two different blocks is probably not useful in achieving more bandwidth.  This was instead of limited the threads all the way down to last power of 2.